src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -17,7 +17,85 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
   const [currentTab, setCurrentTab] = useState<MarkdownEditorTabs>(
     MarkdownEditorTabs.WRITE,
   )
-  const [text, setText] = useState<string>()
+  const [text, setText] = useState<string>('')
+  const [selectionStart, setSelectionStart] = useState<number>(0)
+  const commandList = [
+    'title',
+    'bold',
+    'italic',
+    'strikethrough',
+    'code',
+    'latex',
+    'subscript',
+    'superscript',
+    'link',
+    'image',
+  ]
+
+  const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
+    const input = textAreaRef.current
+    if (input) {
+      input.setSelectionRange(selectionStart, selectionStart)
+    }
+  }, [textAreaRef, selectionStart])
+
+  const handleCommands = (command: string) => {
+    const textVal = textAreaRef.current
+    if (textVal) {
+      const start = textVal.selectionStart
+      const end = textVal.selectionEnd
+      const selected = text.substring(start, end)
+      const first = text.substring(0, start)
+      const later = text.substring(end, text.length)
+
+      let openSyntax = ''
+      let closeSyntax = ''
+      if (command === 'bold') {
+        openSyntax = closeSyntax = '**'
+      } else if (command == 'italic') {
+        openSyntax = closeSyntax = '_'
+      } else if (command === 'strikethrough') {
+        openSyntax = closeSyntax = '--'
+      } else if (command === 'latex') {
+        openSyntax = '$$\\('
+        closeSyntax = '\\)$$'
+      } else if (command === 'subscript') {
+        openSyntax = closeSyntax = '~'
+      } else if (command === 'superscript') {
+        openSyntax = closeSyntax = '^'
+      } else if (command === 'title') {
+        openSyntax = '###'
+      } else if (command === 'link') {
+        openSyntax = '['
+        closeSyntax = '](url)'
+      } else if (command === 'image') {
+        openSyntax = '!['
+        closeSyntax = '](image-url)'
+      } else if (command == 'code') {
+        openSyntax = closeSyntax = '```'
+        const newTextList = `${first}
+        ${openSyntax}
+        ${selected}
+        ${closeSyntax}
+        ${later}`.split('\n')
+        // removes first empty line
+        newTextList.shift()
+        const newText = newTextList.map(s => s.trim()).join('\n')
+        setText(newText)
+        textAreaRef.current.focus()
+        // adds one due to new line
+        setSelectionStart(start + openSyntax.length + 1)
+        return
+      }
+
+      const newText = `${first}${openSyntax}${selected}${closeSyntax}${later}`
+      textAreaRef.current.focus()
+      setSelectionStart(start + openSyntax.length)
```

Comment:
Can you just do this and get rid of the useEffect and useState? 
```suggestion
      const finalCursorPosition = start + openSyntax.length
      textAreaRef.current.setSelectionRange(finalCursorPosition, finalCursorPosition)
```

Reply:
This was my initial attempt. However when doing so the cursor would still go to the end of the text. Whereas the useEffect would put the cursor right next to the openSyntax.

Reply:
I see; the `setSelectionRange` call needs to happen after the `text` state updates, and `useEffect` ensures that that happens in the right order.